core/arch/arm/kernel/thread_private.h

```diff
@@ -228,6 +228,8 @@ void thread_set_irq_sp(vaddr_t sp);
 void thread_set_fiq_sp(vaddr_t sp);
 #endif /*ARM32*/
 
+unsigned long thread_smc(unsigned long func_id, unsigned long arg);
```

Comment:
How about moving this elsewhere and making it more generic (i.e., accept more arguments), so that it may be used by platform code? We already have a use case as a replacement for https://github.com/OP-TEE/optee_os/blob/master/core/arch/arm/plat-ls/ls_hw_unq_key_a64.S.

Reply:
Sure. I'd still want to keep it in this simple form, but accepting four arguments instead of only two.

The rational behind that being don't solve problems that doesn't exist. If we ever need a more complete function we can think about that then.

Reply:
I agree. Four args is fine.